samza-core/src/main/scala/org/apache/samza/util/CoordinatorStreamUtil.scala

```diff
@@ -163,7 +163,14 @@ object CoordinatorStreamUtil extends Logging {
 
     val oldConfig = coordinatorSystemConsumer.getConfig
     if (resetJobConfig) {
-      val keysToRemove = oldConfig.keySet.asScala.toSet.diff(config.keySet.asScala)
+      var keysToRemove = oldConfig.keySet.asScala.toSet.diff(config.keySet.asScala)
+
+      val jobConfig = new JobConfig(config)
+      if (jobConfig.getAutosizingEnabled) {
+        // If autosizing is enabled, we retain auto-sizing related configs
+        keysToRemove = keysToRemove.filter(configKey => !jobConfig.isAutosizingConfig(configKey))
```

Comment:
Is there a case where you want to intentionally delete an autosizing config? For example, for cleaning up after getting a stable set of parameters.

Reply:
The expectation there is that the user copies the internal config values to respective user configs (e.g., using a tool), and disables autosizing (which will delete these internal configs).